test/join.jl

```diff
@@ -31,6 +31,8 @@ module TestJoin
     @test isequal(join(name, job, on = :ID, kind = :right), right)
     @test isequal(join(name, job, on = :ID, kind = :semi), semi)
     @test isequal(join(name, job, on = :ID, kind = :anti), anti)
+    @test_throws ArgumentError error(join(name, job))
```

Comment:
There is no default for `on::Symbol` https://github.com/cjprybol/DataTables.jl/blob/cjp/alyst-groupby/src/abstractdatatable/join.jl#L232